src/controllers/StaffController.js

```diff
@@ -553,6 +625,580 @@ class StaffController {
       return res.status(400).json({ error: err });
     }
   }
+
+  /**
+   * Retrieves all objects from the specified collection for admin access.
+   * @async
+   * @static
+   * @param {Object} req - The request object.
+   * @param {Object} res - The response object.
+   * @returns {Promise<Object>} The result of retrieving all objects.
+   * @throws {Error} If there is an error during the retrieval process.
+   */
+  static async adminGetAll(req, res) {
```

Comment:
Once again, this method would lead to a security breach. The only Mongoose methods to be run here (outside the `methods` folder) are `find`, `findOne`, or `findById`, to get a single user (staff/student) from the database. The endpoints should never use Mongoose directly to find any other model, multiple models, or perform any other CRUD operations. We've discussed this. . .